Jenkinsfile

```diff
@@ -1,7 +1,7 @@
 pipeline {
 	options {
 		timeout(time: 40, unit: 'MINUTES')
-		buildDiscarder(logRotator(numToKeepStr:'15'))
+		buildDiscarder(logRotator(numToKeepStr: '100', artifactNumToKeepStr: '15', artifactDaysToKeepStr: '99'))
```

Comment:
So far, I did not see reliable behavior with `artifactDaysToKeepStr`, so I'd recommend avoiding it completely.

In this case, the combination of values also does not make much sense.  If you only keep up to 15 builds with artifacts, and you build almost daily, artifacts will rarely get 99 days old. As discussed in https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2653, going with conditional values based on the branch is more robust.

I'd start with:
```
buildDiscarder(logRotator(numToKeepStr: ('master' == env.BRANCH_NAME) ? '50' : '5', artifactNumToKeepStr: '2')
```

And in a few weeks you can check back how often you wanted to access the artifacts of builds that are older than 2 builds. :slightly_smiling_face:

Reply:
> artifacts will rarely get 99 days old

look in JDT.core PRs, nobody cleans it:
![image](https://github.com/user-attachments/assets/f1ca5027-f255-411a-86a0-f1b4bab251a7)
the implementation looks solid
https://github.com/jenkinsci/jenkins/blob/5404f14af18155d62f819279a5debddcbff794c5/core/src/main/java/hudson/tasks/LogRotator.java#L213

Reply:
> And in a few weeks you can check back how often you wanted to access the artifacts of builds that are older than 2 builds.
I know i did it in the past. Not often, but after almost every vacation because things got screwed up.

Reply:
This is because there are still open PRs for them like https://github.com/eclipse-jdt/eclipse.jdt.core/pull/30 . If/when a PR is merged/closed it will be cleaned up.

Reply:
cancel "will be" and use "would be" - nobody in jdt.core does care.

Reply:
ok, i'd say 15 builds on master/beta, 2 builds for PRs

Reply:
> The I-Build logfiles to find out when and why it started to fail

Which log files exactly? The Jenkins log files (e.g. [this one](https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/master/528/console)) or some other logs?

Reply:
> ok, i'd say 15 builds on master/beta, 2 builds for PRs

If you are referring to the number of builds to keep with artifacts, I still recommend a lower setting, but you may test your assumption at your convenience.

Reply:
> Which log files exactly?

https://ci.eclipse.org/releng/job/AutomatedTests/job/ep435I-unit-win32-x86_64-java17/lastSuccessfulBuild/artifact/

Reply:
Hmm. That's a different job though.

Reply:
the plan would be to apply such change to all our builds

Reply:
https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/JenkinsJobs/AutomatedTests/I_unit_win32.groovy#L22